lib/crystalball/map_generator/allocated_objects_strategy/object_lister.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Crystalball
+  class MapGenerator
+    class AllocatedObjectsStrategy
+      # Class to list objects from ObjectSpace
+      class ObjectLister
+        attr_reader :ignored_types
+
+        DEFAULT_IGNORED_TYPES = [
+          NilClass, TrueClass, FalseClass, Numeric, String, Regexp,
+          Range, Array, Hash, Struct, Time, IO
+        ].freeze
+
+        # @param Array[Class] ignored_types classes which will be excluded from list
+        def initialize(ignored_types = DEFAULT_IGNORED_TYPES)
+          @ignored_types = ignored_types
+        end
+
+        # @return Set[Integer] object_id list of allocated objects except ignored
+        def list_ids
+          objects = Set.new
+
+          ObjectSpace.each_object(Object) do |object|
+            next if ignored_types.include?(object.class)
+            objects << object.__id__
+          end
+
+          objects
```

Comment:
what do you think about "marking" objects instead of saving them to a list.
Something like this:
```ruby
         ObjectSpace.each_object(Object) do |object|
            next if ignored_types.include?(object.class)
            object.instance_variable_set(:@crystalball_old_object, true)
          end
#.... rspec example here ....
          ObjectSpace.each_object(Object) do |object|
            next if ignored_types.include?(object.class) || object.instance_variable_get(:@crystalball_old_object)
            objects << object
          end
```
Then you'll be able to remove search logic since every object already has info if he's old or new for us

Reply:
It looks like it works faster. But it is impossible to set instance variable for a frozen object. But we can define empty method for the same purpose